### PR TITLE
Add events

### LINKED ISSUE #44: Add Event model
Add Event model and migration with properties:
id, title, description, date, author_id, project_id

### PR DESCRIPTION
**Pull request by filip373**
resolves #44

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Event < ApplicationRecord
+  belongs_to :author, class_name: 'User'
```
Comment (neonowy) [inline app/models/event.rb]: We've decided to use double quotation mark in this project.

File: app/models/event.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  belongs_to :project
+
+  validates :title, :date, presence: true
```
Comment (tprochniewicz) [inline app/models/event.rb]: maybe add validation for date to check if it is not in past

Comment (neonowy) [inline app/models/event.rb]: Hmm, It'd be nice to have the same validation for `date` as in #49. Maybe move it to a custom validator class?

http://guides.rubyonrails.org/active_record_validations.html#performing-custom-validations

File: app/models/event.rb (hunk 1 above)
Comment (filip373) [inline app/models/event.rb]: Ok, I must have missed the discussion about it.

Comment (neonowy) [inline app/models/event.rb]: Oh, actually I think we had this discussion in the Łódź office. Sorry I didn't tell you.

Comment (filip373) [inline app/models/event.rb]: Ok, now I will know :+1:

File: spec/validators/future_date_validator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+RSpec.describe FutureDateValidator do
+  subject { Validatable.new }
+
+  context "when date is nil" do
+    it "is invalid" do
+      expect(subject).to be_invalid
+    end
+
+    it "adds error message about nil" do
+      subject.validate
```
Comment (neonowy) [inline spec/validators/future_date_validator_spec.rb]: You can add empty line below, it'll look better imho.

File: spec/validators/future_date_validator_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  context "when date is in the past" do
+    before { subject.date = Faker::Time.backward(2) }
+
+    it "is invalid" do
+      expect(subject).to be_invalid
+    end
+
+    it "adds error message about the past date" do
+      subject.validate
```
Comment (neonowy) [inline spec/validators/future_date_validator_spec.rb]: Same here :)

File: spec/validators/future_date_validator_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      expect(subject.errors.messages[:date].first).
```
Comment (neonowy) [inline spec/validators/future_date_validator_spec.rb]: Actually this assertion code (with indentation) has exactly 80 chars, so you don't need to split it into two lines.

Comment (filip373) [inline spec/validators/future_date_validator_spec.rb]: I'd rather keep it this way, even if below 80 chars, it's more readable imho.

File: spec/validators/future_date_validator_spec.rb (hunk 3 above)
Comment (filip373) [inline spec/validators/future_date_validator_spec.rb]: For me it looks better without breaks. I think you shouldn't break method body until it's necessary and here there are only 2 lines so I don't think it needs any breaks.

**Comment by pospieszynski** [comment]
+1

**Comment by tprochniewicz** [comment]
@gotar

**Comment by gotar** [comment]
nice one, just squash and you can merge

**Pull request merged**